### ISSUE TITLE
reload tableview without animation?

### ISSUE THREAD

**Issue by 1994ericseo** [SwiftBond/Bond#257]
I followed the same way the table view bindings are done in the example bond-app. But I want to remove the fade-like animation of each cell. Is there any way I can do this using the .bindTo(tableView) method?

**Comment by yanamura3** [comment]
@1994ericseo 
try to use BNDTableViewProxyDatasource.
```
func tableView(tableView: UITableView, animationForRowAtIndexPaths indexPaths: [NSIndexPath]) -> UITableViewRowAnimation {
  return .None
}

func tableView(tableView: UITableView, animationForRowInSections sections: Set<Int>) -> UITableViewRowAnimation {
  return .None
}
```

**Comment by Broich** [comment]
`BNDTableViewProxyDatasource` is gone in Bond5 - What is the preferred way to do this now?

**Comment by srdanrasic** [comment]
TLDR: Should have been supported out of the box, but I've missed to implement it. I'll have a fix version released probably today.

__Long story__ (Good to know!)

Customizing binding options can now be done with custom `TableViewBond`. 

To understand it, it is best to show how default bindings work. Here is the implementation of `bind(to:)` method:

```swift
@discardableResult
public func bind(to tableView: UITableView, animated: Bool = true, createCell: @escaping (DataSource, IndexPath, UITableView) -> UITableViewCell) -> Disposable {
  return bind(to: tableView, using: SimpleTableViewBond<DataSource>(createCell: createCell))
}
```

So it's an overload that uses something called `SimpleTableViewBond`. `SimpleTableViewBond` is just a wrapper struct over a closure that creates cell:

```swift
private struct SimpleTableViewBond<DataSource: DataSourceProtocol>: TableViewBond {

  let createCell: (DataSource, IndexPath, UITableView) -> UITableViewCell

  func cellForRow(at indexPath: IndexPath, tableView: UITableView, dataSource: DataSource) -> UITableViewCell {
    return createCell(dataSource, indexPath, tableView)
  }
}
```

The important part here is the `TableViewBond` conformance. That protocol is defined like this:

```swift
public protocol TableViewBond {

  associatedtype DataSource: DataSourceProtocol

  var animated: Bool { get }

  func cellForRow(at indexPath: IndexPath, tableView: UITableView, dataSource: DataSource) -> UITableViewCell
  func titleForHeader(in section: Int, dataSource: DataSource) -> String?
  func titleForFooter(in section: Int, dataSource: DataSource) -> String?
}
```

With the defaults:

```swift
extension TableViewBond {

  public var animated: Bool {
    return true
  }

  public func titleForHeader(in section: Int, dataSource: DataSource) -> String? {
    return nil
  }

  public func titleForFooter(in section: Int, dataSource: DataSource) -> String? {
    return nil
  }
}
```

What that means is that you can implement your own custom `TableViewBond` to customise defined parameters. 

You would do something like:

```swift
struct MyBond: TableViewBond {

  func cellForRow(at indexPath: IndexPath, tableView: UITableView, dataSource: DataSource) -> UITableViewCell {
    // create cell here
  }

  var animated: Bool {
    return false
  }
}
```

And then do the binding using an instance of your own bond.

```swift
array.bind(to: tableView, using: MyBond())
```

If you don't like the idea of instantiating cells in your bond type, just wrap the creation closure like the SimpleTableView bond does.

```swift
private struct NonAnimatedBond<DataSource: DataSourceProtocol>: TableViewBond {

  let createCell: (DataSource, IndexPath, UITableView) -> UITableViewCell

  func cellForRow(at indexPath: IndexPath, tableView: UITableView, dataSource: DataSource) -> UITableViewCell {
    return createCell(dataSource, indexPath, tableView)
  }

  var animated: Bool {
    return false
  }
}
```

Now, you can extend a date source signal with a binding method that has additional parameter: animated.

```swift
public extension SignalProtocol where Element: DataSourceEventProtocol, Error == NoError {

  @discardableResult
  public func bind(to tableView: UITableView, animated: Bool, createCell: @escaping (Element.DataSource, IndexPath, UITableView) -> UITableViewCell) -> Disposable {
    if animated {
      return bind(to: tableView, using: SimpleTableViewBond<Element.DataSource>(createCell: createCell))
    } else {
      return bind(to: tableView, using: NonAnimatedBond<Element.DataSource>(createCell: createCell))
    }
  }
}
```

But it'll be easier to just wait for the fix :)

**Comment by srdanrasic** [comment]
Fixed `animated` argument in v5.0.2.

**Comment by igorkulman** [comment]
What is the way to do it with the current version? here is no `var animated: Bool` on `TableViewBond` anymore.
